crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -548,6 +572,251 @@ impl SimpleFaucet {
+                }
+                None => {
+                    // Fail with error logging, so that we can continue, but we should never reach here.
+                    error!("Empty request stored in request consumer queue.");
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    // Function to process the batch send of the mcsp queue
+    pub async fn batch_transfer_gases(&self) -> Result<TransactionDigest, FaucetError> {
+        let Ok(mut req_consumer) = tokio::time::timeout(LOCK_TIMEOUT, self.request_consumer.lock()).await else {
+                error!("Timeout when getting request consumer lock");
+                return Err(FaucetError::Internal("Unable to get request queue lock.".to_string()));
+            };
+
+        let mut requests = Vec::new();
+
+        // First request we just wait for without timeout
+        let first_req = req_consumer.recv().await;
+        match first_req {
+            Some((uuid, address, amounts)) => {
+                requests.push((uuid, address, amounts));
+            }
+            None => {
+                return Err(FaucetError::Internal(
+                    "Request consumer queue has corrupted data.".to_string(),
+                ));
+            }
+        }
+        drop(req_consumer);
+        match timeout(BATCH_TIMEOUT, self.batch_gather(&mut requests)).await {
+            Ok(_) => {}
+            Err(_) => {
+                warn!("Batch timeout elapsed while waiting.");
+            }
+        };
+
+        // Early exiting this function if there is nothing to batch.
+        if requests.is_empty() {
+            return Ok(TransactionDigest::ZERO);
+        }
+        let total_requests = requests.len();
+
+        let gas_cost = self.get_gas_cost().await?;
+        // The UUID here is for the batched request
+        let uuid = Uuid::new_v4();
+        info!(
+            ?uuid,
+            "Batch transfer attemped of size: {:?}", total_requests
+        );
+
+        // This loop is utilized to grab a coin that is large enough for the request
+        loop {
+            let gas_coin_response = self
+                .prepare_gas_coin(requests.len() as u64 + gas_cost, uuid)
+                .await;
+
+            match gas_coin_response {
+                GasCoinResponse::ValidGasCoin(coin_id) => {
+                    let tx_data = self
+                        .build_batch_pay_sui_txn(
+                            coin_id,
+                            requests.clone(),
+                            self.active_address,
+                            gas_cost,
+                        )
+                        .await
+                        .map_err(FaucetError::internal)?;
+
+                    // Because we are batching transactions to faucet, we will just not use a real recipient for
+                    // sui address, and instead just fill it with the ZERO address.
+                    let recipient = SuiAddress::ZERO;
```

Comment:
Can be a follow-up, but we should consider getting rid of the `recipient` field from the WAL -- I believe it's only used for logging.